spanner/google/cloud/spanner/snapshot.py

```diff
@@ -153,14 +151,18 @@ def execute_sql(self, sql, params=None, param_types=None, query_mode=None,
         iterator = api.execute_streaming_sql(
             self._session.name, sql,
             transaction=transaction, params=params_pb, param_types=param_types,
-            query_mode=query_mode, resume_token=resume_token, options=options)
+            query_mode=query_mode, options=options)
 
         self._read_request_count += 1
 
+        restart = functools.partial(
+            api.execute_streaming_sql, self._session.name, sql,
+            params=params, param_types=param_types, query_mode=query_mode)
```

Comment:
you need to also pass in the transaction here.

Reply:
I'm assuming likewise for `options` here.

Reply:
8f47fe2